### PR TITLE
Work with methods without red mistakes.

### PR DESCRIPTION
**Pull request by udodikdmitro**

### CODE REVIEW

File: src/package1/CustomArrayList.java [hunk 1]
```diff
@@ -4,104 +4,216 @@
...
     @Override
-    public boolean isEmpty(List<String> name) {
+    public int size() {
+        int i = 0;
+
+        for (String s : elements) {
+            i++;
+        }
+        return i;
+    }
```
Comment (udodikdmitro) [inline src/package1/CustomArrayList.java]: I'll use variable size instead

File: src/package1/CustomArrayList.java [hunk 2]
```diff
@@ -5,55 +5,46 @@
...
     @Override
     public boolean isEmpty() {
 
-        for (String s : elements) {
-            if (s != null) {
-                return false;
-            }
+        if (size == 0) {
+            return true;
         }
```
Comment (ezseeton) [inline src/package1/CustomArrayList.java]: use return with condition instead

File: src/package1/CustomArrayList.java [hunk 3]
```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
...
+
+        elements[size] = s;
+        size++;
 
-        for (int i = 0; i < name.size(); i++) {
-            a[i] = name.get(i);
+        if (size == elements.length) {
+            String[] a = new String[(int) (elements.length * 1.5)];
+            elements = a;
         }
```
Comment (ezseeton) [inline src/package1/CustomArrayList.java]: forgot to copy previous elements to created array

File: src/package1/CustomArrayList.java [hunk 4]
```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
...
     }
 
+//    Використовувати size замість length
+//    Елементи мають переміщуватись вліво в сторону видаленого елемента.
     @Override
-    public boolean remove(List<String> name, Object o) {
-        for (int i = 0; i < name.size(); i++) {
-            if (name.get(i).equals(o)) {
-                name.remove(name.get(i));
+    public boolean remove(Object o) {
```
Comment (ezseeton) [inline src/package1/CustomArrayList.java:68]: 1. Use size, not length
2. Remove is just shift elements to left

File: src/package1/CustomArrayList.java [hunk 5]
```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
...
-    public boolean containsAll(List<String> name, Collection<String> c) {
-        for (int i = 0; i < name.size(); i++) {
-            for (String s : c) {
+    public boolean containsAll(Collection<?> c) {
+
+        int j = 0;
+
+        for (Object s : c) {
 
+            for (int i = 0; i < elements.length; i++) {
```
Comment (ezseeton) [inline src/package1/CustomArrayList.java]: size

File: src/package1/CustomArrayList.java [hunk 6]
```diff
@@ -33,75 +40,162 @@ public boolean contains(List<String> name, Object o) {
...
     }
 
+    //Використовувати size.
     @Override
-    public boolean containsAll(List<String> name, Collection<String> c) {
-        for (int i = 0; i < name.size(); i++) {
-            for (String s : c) {
+    public boolean containsAll(Collection<?> c) {
+
+        int j = 0;
```
Comment (ezseeton) [inline src/package1/CustomArrayList.java]: use meaningful names of variables. For example "amountOfMatchs" instead of j

**Review by ezseeton — CHANGES_REQUESTED** [review]
Please use size everywhere in code. Use meaningful variable names

File: src/package1/CustomArrayList.java (hunk 1 above)
Comment (udodikdmitro) [reply src/package1/CustomArrayList.java]: Done

File: src/package1/CustomArrayList.java (hunk 2 above)
Comment (udodikdmitro) [reply src/package1/CustomArrayList.java]: Changed everywhere.

File: src/package1/CustomArrayList.java (hunk 3 above)
Comment (udodikdmitro) [reply src/package1/CustomArrayList.java]: Now it copies previous elements.

File: src/package1/CustomArrayList.java (hunk 4 above)
Comment (udodikdmitro) [reply src/package1/CustomArrayList.java:68]: Fixed.

File: src/package1/CustomArrayList.java (hunk 5 above)
Comment (udodikdmitro) [reply src/package1/CustomArrayList.java]: Now iterator() is using.

File: src/package1/CustomArrayList.java (hunk 6 above)
Comment (udodikdmitro) [reply src/package1/CustomArrayList.java]: Changes names.

**Review by ezseeton — APPROVED** [review]

**Pull request merged**